crates/payload/basic/src/lib.rs

```diff
@@ -63,6 +64,7 @@ pub struct BasicPayloadJobGenerator<Client, Tasks, Builder> {
     builder: Builder,
     /// Stored `cached_reads` for new payload jobs.
     pre_cached: Option<PrecachedState>,
+    pub is_resolving: Arc<AtomicBool>,
```

Comment:
do we need this pub?

let's also add a oneliner doc

Reply:
To enable other parts of the system to utilize it ,  doc added